symengine/expression.h

```diff
@@ -179,6 +179,22 @@ inline Expression expand(const Expression &arg)
     return expand(arg.get_basic());
 }
 
+template <typename T, typename U,
+          typename = enable_if_t<std::is_same<U, Expression>::value
+                                 and std::is_same<T, Expression>::value>>
+inline bool unified_eq(const T &a, const U &b)
+{
+    return a == b;
+}
+
+template <typename T, typename U,
+          typename = enable_if_t<std::is_same<U, Expression>::value
+                                 and std::is_same<T, Expression>::value>>
+inline int unified_compare(const T &a, const U &b)
+{
+    return unified_compare(a.get_basic(), b.get_basic());
+}
+
```

Comment:
Would
```
inline bool unified_eq(const Expression &a, const Expression &b)
```
be a better choice?
Pro: 
 - Reduce compile time

Possible behavior change:
 - Overloading precedence: non-template function > template function. Changing it to non-template version might cause unexpected behavior when you want to match a subclass of `Expression` with template. In that case this version will have higher priority.